src/components/LoginButton.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from 'react';
+import firebase from 'firebase';
+import Button from '@material-ui/core/Button';
+import LoginDialog from './LoginDialog';
+
+export default function LoginButton() {
+  const { currentUser } = firebase.auth(); // determines how to initialize isSignedIn
+  const [isSignedIn, setLoggedIn] = useState(!!currentUser);
+  const [loginDialogOpen, setDialogOpen] = useState(false); // initialized to false
+
+  firebase.auth().onAuthStateChanged((user) => {
+    if (user) {
+      setLoggedIn(true);
+      setDialogOpen(false);
+    } else {
+      setLoggedIn(false);
+    }
+  });
+
+  const signUserOut = () => firebase.auth().signOut();
+  const showLoginDialog = () => setDialogOpen(true);
+
+  return (
+    <div>
+      <Button
+        variant="outlined"
+        color="secondary"
+        onClick={isSignedIn ? signUserOut : showLoginDialog}
+      >
+        {isSignedIn ? 'Sign out' : 'Login'}
+      </Button>
+      <LoginDialog open={loginDialogOpen} onClose={false} />
```

Comment:
`onClose` probably shouldn't be false. The dialog should probably close when the user tries to close the dialog without logging in.

Reply:
Yeah I ran into this issue that it does not close if the user does not want to login, any recommendation for how to fix?

Reply:
Instead of `false` pass it a function that'll close the login dialog

Reply:
Ok nice, I passed in this function -> `const handleClose = () => setLoginDialogOpen(false);` and it now works when you click outside the dialog it'll close